TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/HeaderAdderHandler.java

```diff
@@ -0,0 +1,52 @@
+package org.opensearch.migrations.trafficcapture.proxyserver.netty;
+
+import java.nio.charset.StandardCharsets;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+
+public class HeaderAdderHandler extends ChannelInboundHandlerAdapter {
+    private static final ByteBuf CRLF_BYTE_BUF =
+        Unpooled.unreleasableBuffer(Unpooled.wrappedBuffer("\r\n".getBytes(StandardCharsets.UTF_8)));
+    private static final ByteBuf LF_BYTE_BUF =
+        Unpooled.unreleasableBuffer(Unpooled.wrappedBuffer("\n".getBytes(StandardCharsets.UTF_8)));
+    boolean insertedHeader = false;
+    private final ByteBuf headerLineToAdd;
+    boolean useCarriageReturn;
+
+    public HeaderAdderHandler(ByteBuf headerLineToAdd) {
+        this.headerLineToAdd = headerLineToAdd.retain();
+    }
+
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+        if (!(msg instanceof ByteBuf) || insertedHeader) {
+            super.channelRead(ctx, msg);
+            return;
+        }
+        var buf = (ByteBuf) msg;
+        buf.markReaderIndex();
```

Comment:
This is the only spot that we're using it.  The full documentation on mark & reset from the ByteBuf javadoc is 4 sentences long.  Considering that this prevents the code from having to alloc a new object, yes, I'd say that it's worth it.

Reply:
Actually - there's a bug that I hadn't accounted for - we can remove payload content that matches the pattern.  I'll put a test in to prevent that & fix it.